backend/src/test/java/com/google/models/TimestampTest.java

```diff
@@ -0,0 +1,52 @@
+package com.google.models;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertFalse;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+
+/** Contain tests for methods in {@link Timestamp} class. */
+@RunWith(JUnit4.class)
+public final class TimestampTest {
+  private static final int DAY_CONST = 1;
+  private static final int MONTH_CONST = 1;
+  private static final int YEAR_CONST = 2000;
+  private static final Timestamp timestamp = new Timestamp(DAY_CONST, MONTH_CONST, YEAR_CONST);
+
+  @Test
+  public void constructor_convertStringToDate() throws Exception {
+    Timestamp stringTimestamp = new Timestamp("2020-12-5");
+
+    assertEquals(stringTimestamp.getDay(), 5);
+    assertEquals(stringTimestamp.getMonth(), 12);
+    assertEquals(stringTimestamp.getYear(), 2020);
+  }
+
+  @Test(expected = Exception.class)
+  public void constructor_throwsExceptionForBadFormat() throws Exception {
+    Timestamp stringTimestamp = new Timestamp("202012-5");
```

Comment:
Include other wrong / bad strings? more than just this one, and not sure about this but maybe check the error statements / check that the right errors are being logged
e.g. length of string is wrong

Reply:
thanks for pointing this out! i added more tests and fix code to make `timestamp` object more flexible in detecting errors.